test/test_optim.py

```diff
@@ -877,10 +868,10 @@ def test_fused_optimizers(self):
         optimizer_pairs_with_flags = tuple(itertools.product(
             (optim.Adam, optim.AdamW),
             (
-                dict(weight_decay=1., amsgrad=False),
-                dict(weight_decay=1., amsgrad=True),
-                dict(weight_decay=0., amsgrad=False),
-                dict(weight_decay=0., amsgrad=True),
+                dict(weight_decay=1., amsgrad=False, foreach=False),
```

Comment:
Why must we add these?

Reply:
not must. it's just my taste

Reply:
The change here is that we would be testing against the for-loop versus the foreach implementation (the new default). Is that what you are implying?